### PR TITLE
wiz: fix state retention between iterations

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Fix state retention between iterations and clean up CEL code structure.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] Best reviewed commit-wise. Notes for particularly subtle changes will be added, but may be hidden by GitHub UI if viewed as a full set of changes.
- [ ] First commit best reviewed with ignore whitespace changes.

## Related issues

-

### CODE REVIEW

File: packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -61,27 +61,21 @@ program: |
...
                   )
-                :
-                  (
-                    has(state.cursor) && has(state.cursor.first_timestamp) && state.cursor.first_timestamp != null
-                    ?
-                      state.cursor.first_timestamp
-                    :
-                      null
-                  )
               )
```
Comment (efd6) [inline packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs]: This logic is subtle, so please take extra care here. Similar in the other two data streams.

File: packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs [hunk 2]
```diff
@@ -64,7 +64,7 @@ program: |
           "after": state.?end_cursor.value.orValue(null),
           "filterBy": {
             "timestamp": {
-              "after": state.?want_more.orValue(true) ?
+              "after": state.want_more ?
```
Comment (efd6) [inline packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs:67]: The logic here flows algebraically, but the condition is odd IMO. Please take extra care here.

File: packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs [hunk 3]
```diff
@@ -53,85 +53,68 @@ state:
...
-                       state.cursor.first_timestamp
-                       :
-                       null
-                     )
-                ) }
-                }
+              "after": state.want_more ?
+                  state.?cursor.first_timestamp.orValue(null)
+                :
+                  state.?cursor.last_timestamp.orValue(string(now() - duration(state.initial_interval)))
```
[the remark this answers is not in the document]
Comment (efd6) [reply packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs:70]: I can revert that commit if necessary. The documentation for wiz is not public, so I can't check. They are both RFC 3339.

[the remark this answers is not in the document]
Comment (kcreddy) [reply packages/wiz/data_stream/audit/agent/stream/cel.yml.hbs:70]: No, thats fine. I was curious if it was something missed.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Comment by efd6** [comment]
Waiting on internal testing against a real API.

**Comment by efd6** [comment]
Tested successfully against a real endpoint, so merging.

**Pull request merged**